runtime/compiler/z/codegen/J9S390PrivateLinkage.cpp

```diff
@@ -1531,7 +1531,8 @@ TR::S390PrivateLinkage::buildVirtualDispatch(TR::Node * callNode, TR::RegisterDe
             char *j2iSignature = fej9->getJ2IThunkSignatureForDispatchVirtual(methodSymbol->getMethod()->signatureChars(), methodSymbol->getMethod()->signatureLength(), comp());
             int32_t signatureLen = strlen(j2iSignature);
             virtualThunk = fej9->getJ2IThunk(j2iSignature, signatureLen, comp());
-            if (!virtualThunk)
+            // For remote compilations we always need to regenerate the thunk inside the code cache for this compilation.
+            if (!virtualThunk || comp()->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER)
```

Comment:
The test for `comp()->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER` is gone on x86 so we can completely eliminate this change.